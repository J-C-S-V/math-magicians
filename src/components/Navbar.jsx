import { Outlet } from 'react-router-dom';
import '../styles/layout.scss';

export default function Navbar() {
  return (
    <>
      <nav data-testid="navbar" className="navbar">
        <h1 className="navbar__h1">Math Magicians</h1>
        <ul className="navbar__ul">
          <li className="navbar__li">
            <a className="navbar__a" href="http://localhost:3000/">
              Home
            </a>
          </li>
          <li className="navbar__li">
            <a className="navbar__a" href="http://localhost:3000/calculator">
              Calculator
            </a>
          </li>
          <li className="navbar__li">
            <a className="navbar__a" href="http://localhost:3000/quote">
              Quote
            </a>
          </li>
        </ul>
      </nav>
      <Outlet />
    </>
  );
}
