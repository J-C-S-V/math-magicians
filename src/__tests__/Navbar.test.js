import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Navbar from '../components/Navbar';

describe('Navbar component', () => {
  test('renders', () => {
    expect(<Navbar />).toMatchSnapshot();
  });
  test('renders the Navbar component', () => {
    render(<Navbar />);
    const navbarElement = screen.getByTestId('navbar');
    expect(navbarElement).toBeInTheDocument();
  });

  test('renders the Math Magicians title', () => {
    render(<Navbar />);
    const titleElement = screen.getByText('Math Magicians');
    expect(titleElement).toBeInTheDocument();
  });

  test('renders the Home navigation link', () => {
    render(<Navbar />);
    const homeLink = screen.getByText('Home');
    expect(homeLink).toBeInTheDocument();
  });
});
