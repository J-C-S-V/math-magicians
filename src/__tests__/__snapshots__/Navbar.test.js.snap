// Jest Snapshot v1, https:
//goo.gl/fbaqlp
exports[`navbar component renders 1`] = `<navbar />`;JestSnapshotv1,https
